### PR TITLE
Add bang operator to operators table

### PR DESCRIPTION
**Pull request by thiagola92**
Add `!` (bang) to operators table in the `not` row.  

```gdscript
var is_online = true
is_online = !is_online
```

Bang token: https://github.com/godotengine/godot/blob/3bd74cd67bfc5484b3f5d4b47da66c55457474c7/modules/gdscript/gdscript_tokenizer.h#L69  

Bang does the same as `not`: https://github.com/godotengine/godot/blob/6debf86d516f44678730be7621784a3883ab5140/modules/gdscript/gdscript_parser.cpp#L2414-L2415

### CODE REVIEW

**Comment by YuriSizov** [comment]
Aliases for `and` and `or` are not listed in this table either. This is probably because the style guide insists on using spelled out operators rather than their symbolic aliases. I wouldn't mind adding them here, but it should be done for all of them, not just for negation.

**Comment by thiagola92** [comment]
There's no situation where this operators are recommended? I don't want to encourage people to use something that goes against the style guide in all situations... But i want to document every feature.  

I though about bang operator when flipping the boolean value because `not` operator sounded strange when reading:  
```gdscript
is_active = not is_active
```

Anyway, it may be a bad idea to add... don't know.  

Note: Style guide also doesn't mention bang operator  
https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_styleguide.html#boolean-operators

**Comment by Calinou** [comment]
I think these should be mentioned, but make sure to link to the style guide when doing so.

**Comment by thiagola92** [comment]
@Calinou Mentioning this directly in the table or in the style guide?

File: tutorials/scripting/gdscript/gdscript_basics.rst [hunk 1]
```diff
@@ -268,16 +268,19 @@ The following is the list of supported operators and their precedence.
...
+|                                                                                | standalone expression)                    |
 |                                                                                |                                           |
 |                                                                                | Content iterator (when used with the      |
-|                                                                                | for_ keyword).                            |
+|                                                                                | for_ keyword)                             |
 +--------------------------------------------------------------------------------+-------------------------------------------+
-| ``not``                                                                        | Boolean NOT                               |
+| ``not`` ``!``                                                                  | Boolean NOT and it's                      |
+|                                                                                | :ref:`aliase<boolean_operators>`          |
```
Comment (mhilbrunner) [inline tutorials/scripting/gdscript/gdscript_basics.rst:277]:
```suggestion
| ``not`` ``!``                                                                  | Boolean NOT and its                       |
|                                                                                | :ref:`aliases<boolean_operators>`          |
```

File: tutorials/scripting/gdscript/gdscript_basics.rst [hunk 2]
```diff
@@ -268,16 +268,19 @@ The following is the list of supported operators and their precedence.
...
 +--------------------------------------------------------------------------------+-------------------------------------------+
-| ``and``                                                                        | Boolean AND                               |
+| ``and`` ``&&``                                                                 | Boolean AND and it's                      |
+|                                                                                | :ref:`aliase<boolean_operators>`          |
```
Comment (mhilbrunner) [inline tutorials/scripting/gdscript/gdscript_basics.rst:280]:
```suggestion
| ``and`` ``&&``                                                                 | Boolean AND and its                       |
|                                                                                | :ref:`aliases<boolean_operators>`         |
```

File: tutorials/scripting/gdscript/gdscript_basics.rst [hunk 3]
```diff
@@ -268,16 +268,19 @@ The following is the list of supported operators and their precedence.
...
 +--------------------------------------------------------------------------------+-------------------------------------------+
-| ``or``                                                                         | Boolean OR                                |
+| ``or`` ``||``                                                                  | Boolean OR and it's                       |
+|                                                                                | :ref:`aliase<boolean_operators>`          |
```
Comment (mhilbrunner) [inline tutorials/scripting/gdscript/gdscript_basics.rst:283]:
```suggestion
| ``or`` ``||``                                                                  | Boolean OR and its                        |
|                                                                                | :ref:`aliases<boolean_operators>`         |
```

**Review by mhilbrunner — APPROVED** [review]

File: tutorials/scripting/gdscript/gdscript_basics.rst [hunk 4]
```diff
@@ -268,16 +268,19 @@ The following is the list of supported operators and their precedence.
...
+|                                                                                | standalone expression)                    |
 |                                                                                |                                           |
 |                                                                                | Content iterator (when used with the      |
-|                                                                                | for_ keyword).                            |
+|                                                                                | for_ keyword)                             |
 +--------------------------------------------------------------------------------+-------------------------------------------+
-| ``not``                                                                        | Boolean NOT                               |
+| ``not`` ``!``                                                                  | Boolean NOT and its                       |
+|                                                                                | :ref:`aliases<boolean_operators>`          |
```
Comment (mhilbrunner) [inline tutorials/scripting/gdscript/gdscript_basics.rst:277]:
```suggestion
|                                                                                | :ref:`aliases<boolean_operators>`         |
```

**Comment by mhilbrunner** [comment]
Thanks for contributing! - Merged.

**Pull request merged**